### ISSUE TITLE
Why fine-tuning in single-task setting (not as stated in the paper)?

### ISSUE THREAD

**Issue by svboeing** [namisan/mt-dnn#7]
In the arxiv paper it is stated:
> In the multi-task fine-tuning stage, we use minibatch based stochastic gradient descent (SGD) to
learn the parameters of our model (i.e., the parameters of all shared layers and task-specific layers) as shown in Algorithm 1. In each epoch, a
mini-batch b_t
is selected(e.g., among all 9 GLUE
tasks), and the model is updated according to the
task-specific objective for the task t. This approximately optimizes the sum of all multi-task objectives.

If I understand it correctly, in your code this multi-task fine-tuning stage is called `MTL refinement`. Then why do you fine-tune for each task in single task setting in your `fine-tuning` stage? There is no such stage in the original paper.
Also, in `run_mt_dnn.sh` there are lines:
`train_datasets="mnli,rte,qqp,qnli,mrpc,sst,cola,stsb"
test_datasets="mnli_matched,mnli_mismatched,rte"`
Why do you only test on `mnli` and `rte` and not test on all other tasks?
I would also like to ask if I can switch from `BERT large` to `BERT base` there because i only have one 1080 GTX card.

Thank you.

**Comment by namisan** [comment]
The paper attached is slightly outdated and it tested on old version of GLUE (please refer GLUE website for detailed information). If you look into the tasks in GLUE, pair-wised sentence tasks, e.g, NLI, dominates. Thus, the single sentence task is under training. It requires another step of fine-turning to obtain SOTA on the leaderboard.  We select models based on MNLI/RTE. Of course, you can test on all the tasks. You can switch to _BERT base_, if you don't have powerful GPUs.
As mentioned in the README, we will update the paper.

**Comment by colfire** [comment]
I just wonder if we can obtain or approach SOTA on the leaderboard only using feature-based fine-tuning on the single task, instead of the end-to-end fine-tuning. 
Note: the feature-based fine-tuning and end-to-end fine-tuning are all based on the MT-DNN.

**Comment by namisan** [comment]
This is a good question. However, I don't know the answer. At first stage, we tried the feature-based approach and find it didn't help. Thus, we haven't explored this direction, but I still believe that it is worth to give a shot.

**Comment by colfire** [comment]
Thanks for your reply.
By using the feature-based approach, we can significantly reduce the inference cost for multiple tasks.

**Comment by ConanCui** [comment]
> I just wonder if we can obtain or approach SOTA on the leaderboard only using feature-based fine-tuning on the single task, instead of the end-to-end fine-tuning.
> Note: the feature-based fine-tuning and end-to-end fine-tuning are all based on the MT-DNN.
Sorry, I am wondering what‘s the definition about `feature-based fine-tuning` and 'end-to-end fine-tuning', and what's the difference between them?

**Comment by colfire** [comment]
Conan, you can the definition in original Bert paper for your reference :)
https://arxiv.org/pdf/1810.04805.pdf

**Comment by agupta74** [comment]
Are the MT-DNN results (shown in the image below) for each GLUE task in the paper based on using a single multi-task model or do you fine-tune on each of the GLUE tasks (as specified in this Git repo) on top of the multi-task model?

<img width="711" alt="image" src="https://user-images.githubusercontent.com/21690396/57394395-49d2a080-717a-11e9-9d34-fc9d2c58981b.png">
